app/controllers/casebooks/sections_controller.rb

```diff
@@ -0,0 +1,121 @@
+require 'net/http'
+require 'uri'
+
+class Casebooks::SectionsController < ApplicationController
+  layout 'casebooks'
+  before_action :find_book
+  before_action :find_section, except: [:new, :create, :index]
+  before_action :find_parent, only: [:new, :create]
+  before_action :canonical_redirect, only: [:show]
+  before_action :disable_turbolinks_cache, only: [:new]
```

Comment:
I think this isn't used presently because clicking 'new section' hits `create` and goes straight to the edit state for the section. The issue here was that if you go to the new section route, and that view renders the full table of contents, you'll see a flash of content from the previous hit to new section. I.e. you'll see the last section you added briefly become editable before it actually adds a new section and makes that one editable.

Solved by disabling cache, but this would also be fixed by adding a little more advanced JS so that new routes happen mostly client-side. I might prefer to do that.